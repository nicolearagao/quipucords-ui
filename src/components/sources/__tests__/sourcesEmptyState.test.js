import React from 'react';
import SourcesEmptyState from '../sourcesEmptyState';

describe('SourcesEmptyState Component', () => {
  it('should render a basic component', async () => {
    const props = {};

    const component = await shallowComponent(<SourcesEmptyState {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should render the application name', () => {
    const props = {
      uiShortName: 'Ipsum'
    };

    const component = renderComponent(<SourcesEmptyState {...props} />);
    expect(component.getByRole('heading')).toMatchSnapshot('application name');
  });
});
