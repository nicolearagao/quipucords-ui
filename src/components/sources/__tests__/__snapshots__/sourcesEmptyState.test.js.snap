// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourcesEmptyState Component should render a basic component: basic 1`] = `
<EmptyState
  className="quipucords-empty-state"
  variant="large"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h1"
  >
    t(view.empty-state, {"context":"title","name":"Quipucords"})
  </Title>
  <EmptyStateBody>
    t(view.empty-state, {"context":"description"})
  </EmptyStateBody>
  <EmptyStatePrimary>
    <Button
      onClick={[Function]}
    >
      t(view.empty-state, {"context":"label"})
    </Button>
  </EmptyStatePrimary>
</EmptyState>
`;

exports[`SourcesEmptyState Component should render the application name: application name 1`] = `
<h1
  class="pf-c-title pf-m-2xl"
  data-ouia-component-id="OUIA-Generated-Title-1"
  data-ouia-component-type="PF4/Title"
  data-ouia-safe="true"
>
  t(view.empty-state, {"context":"title","name":"Ipsum"})
</h1>
`;
